code/datums/quirks/_quirk.dm

```
@@ -67,6 +69,12 @@
 	if(quirk_holder)
 		CRASH("Attempted to add quirk to a holder when it already has a holder.")
 
+	//BUBBER EDIT ADDITION START - Species quirks
+	//We silently fail if a whitelisted quirk *somehow* got into someone's preferences.
+	if(quirk_whitelist_flags & QUIRK_SLIMEPERSON_ONLY && !isroundstartslime(new_holder))
+		return
```

```suggestion
		CRASH("Attempted to add quirk [name] to [quirk_holder], but the species is ineligible for it!")
```

---

I'm fairly sure this just runtimes, i. e. returns, and we ought to fail loudly.

---

https://www.byond.com/docs/ref/#/proc/CRASH